src/main/java/com/minecolonies/coremod/colony/ColonyManager.java

```diff
@@ -149,9 +168,146 @@ public static Colony createColony(@NotNull final World w, final BlockPos pos, @N
 
+        ColonyManager.notifyChunksInRange(colony.getWorld(), true, colony.getID(), colony.getCenter(), colony.getDimension());
         return colony;
     }
 
+    /**
+     * Notify all chunks in the range of the colony about the colony.
+     * @param world the world of the colony.
+     * @param add remove or add
+     */
+    public static void notifyChunksInRange(final World world, final boolean add, final int id, final BlockPos center, final int dimension)
+    {
+        final Chunk centralChunk = world.getChunkFromBlockCoords(center);
+        if(centralChunk.getCapability(CLOSE_COLONY_CAP, null).getOwningColony() == id && add)
+        {
+            return;
+        }
+        final IColonyTagCapability cap = centralChunk.getCapability(CLOSE_COLONY_CAP, null);
+
+        if(add)
+        {
+            cap.setOwningColony(id);
+            cap.addColony(id);
+        }
+        else
+        {
+            cap.removeColony(id);
+        }
+        centralChunk.markDirty();
+        MineColonies.getNetwork().sendToAll(new UpdateChunkCapabilityMessage(centralChunk.getCapability(CLOSE_COLONY_CAP, null), centralChunk.x, centralChunk.z));
+
+        final int chunkX = centralChunk.x;
+        final int chunkZ = centralChunk.z;
+
+        final int range = Configurations.gameplay.workingRangeTownHallChunks;
+        final int buffer = Configurations.gameplay.townHallPaddingChunk;
+
+        final int maxRange = range * 2 + buffer;
+
+        for(int i = chunkX - maxRange; i <= chunkX + maxRange; i++)
+        {
+            for(int j = chunkZ - maxRange; j <= chunkZ + maxRange; j++)
+            {
+                final BlockPos pos = new BlockPos(i, 0, j);
+                if(i >= chunkX - range && j >= chunkZ - range && i <= chunkX + range && j <= chunkZ + range)
+                {
+                    ownedChunks.put(new ChunkPos(i,j,dimension), new ChunkLoadStorage(id, pos, add, dimension));
+                }
+                else
+                {
+                    closeChunks.add(new ChunkLoadStorage(id, pos, add, dimension));
+                }
+            }
+        }
+        removedElements = RELOAD_AFTER_X_ELEMENTS;
+    }
+
+    /**
+     * Load each chunk on world tick.
+     * @param world the world.
+     */
+    private static void loadChunkAndNotify(final World world)
+    {
+        if(!ownedChunks.isEmpty())
+        {
+            final ChunkPos storage = ownedChunks.keySet().iterator().next();
+            loadChunk(world, storage.getX(), storage.getZ(), storage.getDim());
+        }
+        else if(!closeChunks.isEmpty())
+        {
+            final ChunkLoadStorage storage = closeChunks.get(0);
+            if(world.provider.getDimension() != storage.getDimension())
+            {
+                return;
+            }
+            final BlockPos idNow = storage.getPos();
+            if (removedElements % 100 == 0)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 100 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)